mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -55,6 +58,52 @@ void createAttnTuningRangeBF(TuningParamSet *newSpace, AttentionOp attnOp,
   }
 }
 
+auto getOptinalSplitKFactors(RockGemmWrapperInterface gemmOp,
```

Comment:
No auto on return types please.

And perhaps `SmallVectorImpl<int64_t>& factors`

Reply:
Ok. But let's not return a reference to a SmallVector.

Reply:
A lot of functions take `SmallVectorImpl<T>& out` as a parameter so they can explicitly write onto tho caller's stack.

Reply:
Ok. I though we were talking about the return parameter. I made it `SmallVector`. I usually rely on the return value optimization.